fullstack-javascript/readme.md

```diff
@@ -1 +1,70 @@
-## TBFH
+## frontend-task
+> Build a GitHub repository search application
+
+This case study aims to assess how you approach a problem starting from the high level solution to the low level implementation details and code quality. You will not be penalized for asking questions, so don't hesitate to ask us if you need any clarification.
+
+## The Task
+
+You are required to build a fullstack single page application built with
+
+* React.js
+* TypeScript
+* ExpressJs, REDIS
+* Redux and [redux-persist](https://github.com/rt2zz/redux-persist)
+* [React Router](https://github.com/ReactTraining/react-router)
+* Vanilla CSS, Sass, Styled Components or any other CSS-in-JS but no frameworks allowed.
+
+
+## Backend Requirements
+
+Write an "Search" Backend API endpoint which eventually collect the data from Github & stores it in REDIS.
+
+Create two API Endpoints:
+1.  "/api/search"
+    - Receives a POST request with search type(**users** or **repositories** or **issues**) & search text(mandatory).
+    - The results will be fetched from the GitHub API & cache it for atleast 2 hours.
+    * [GitHub Search API Docs](https://developer.github.com/v3/search/)
+2.  "/api/search/clearcache" : Clear Backend Caching
+
+Here are some of the items that you should take care of
+
+* Add Caching so that the same request is not called again.
+* Write Swagger documentation with clear description, request, response & example for the endpoints
+* Write API Flows for unit testing the API using any assertion library such as "SuperTest"
```

Comment:
Move the backend after the frontend so that the candidate understand what they need to build.

Reply:
Also no need for the API flows

Reply:
Make sense. Thanks @kamranahmedse. Moving BE after FE. Adding the API flow as an Bonus/Optional